extensions/api/data-management/contractdefinition/src/test/java/org/eclipse/dataspaceconnector/api/datamanagement/contractdefinition/ContractDefinitionApiControllerTest.java

```diff
@@ -0,0 +1,121 @@
+package org.eclipse.dataspaceconnector.api.datamanagement.contractdefinition;
+
+import org.eclipse.dataspaceconnector.spi.monitor.Monitor;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+
+class ContractDefinitionApiControllerTest {
+    private ContractDefinitionApiController controller;
+
+    @BeforeEach
+    void setup() {
+        var monitor = mock(Monitor.class);
+        controller = new ContractDefinitionApiController(monitor);
+    }
+
+    @Test
+    void getAll_paging_noFilter() {
+        //todo: implement
+    }
+
+    @Test
+    void getAll_paging_pageSizeTooLarge() {
+        //todo: implement
+    }
+
+    @Test
+    void getAll_paging_offsetOutOfBounds() {
+        //todo: implement
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = { "id=id1", "id = id1", "id =id1", "id= id1" })
+    void getAll_paging_withValidFilter(String filter) {
+        //todo: implement
+    }
+
+
+    @ParameterizedTest
+    @ValueSource(strings = { "id > id1", "id < id1", "id like id1", "id inside id1" })
+    void getAll_paging_filterWithInvalidOperator(String filter) {
+        //todo: implement
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = { "id>id1", "id<id1" })
```

Comment:
Is there a particular reason why you moved this out to a separate test? They are the same operators as in the above test or not?

Reply:
it's a subtle difference, but in above variant there are spaces, making it valid, whereas this is invalid. If you look at the `QuerySpec`, it'll make sense.

Reply:
Understood. So for example instead of `"id>id1"` there could also be something like `idlikeid1` 😄  You just choose `<`, `>` because those are "more" real operators. It's more an example/test that there MUST be spaces inside the filter. But then why are there some with spaces and some without in line 37 - shouldn't it be consistent across "all" operators (I know there is currently only the `=` as an operator)?

Other than that, it's fine for me, and you can leave it as is.

Reply:
The equality operator is a bit special, because it is parsed explicitly and there is an option to translate it to a "contains" operator automatically, assuming that that's probably far more common than an accurate `equals`. 

However, the more I think about this, the less merit I see in this. I may change this in another PR.